File: Java/Objeto.java
package Java;

class Objeto {
    private final String nombre;
    private boolean contaminado;

    public Objeto(String nombre) {
        if (nombre == null || nombre.isBlank()) {
            throw new IllegalArgumentException("El nombre del objeto no puede estar vacio.");
        }

        this.nombre = nombre;
        this.contaminado = false;
    }

    public String getNombre() {
        return nombre;
    }

    public boolean estaContaminado() {
        return contaminado;
    }

    void contaminar() {
        contaminado = true;
    }

    void desinfectar() {
        contaminado = false;
    }

    public void mostrarEstado() {
        String estado = contaminado ? "CONTAMINADO" : "LIMPIO";
        System.out.printf("Objeto:  %-22s | Estado: %s%n", nombre, estado);
    }
}

File: CPP/Objeto.cpp
#include "Objeto.h"
#include <iostream>
using namespace std;

Objeto::Objeto(string nombre) {
    this->nombre = nombre;
    this->contaminado = false;
}

string Objeto::getNombre() {
    return nombre;
}

bool Objeto::estaContaminado() {
    return contaminado;
}

void Objeto::contaminar() {
    contaminado = true;
}

void Objeto::desinfectar() {
    contaminado = false;
}

void Objeto::mostrarEstado() {
    cout << "Objeto: " << nombre
         << " | Estado: "
         << (contaminado ? "CONTAMINADO" : "LIMPIO")
         << endl;
}